test/e2e/common/pods.go

```diff
@@ -873,6 +879,154 @@ var _ = framework.KubeDescribe("Pods", func() {
+		_, err = f.ClientSet.CoreV1().Pods(testNamespaceName).Patch(context.TODO(), testPodName, types.StrategicMergePatchType, []byte(podPatch), metav1.PatchOptions{})
+		framework.ExpectNoError(err, "failed to patch Pod %s in namespace %s", testPodName, testNamespaceName)
+		ctx, cancel = context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+		_, err = watchtools.Until(ctx, podsList.ResourceVersion, w, func(event watch.Event) (bool, error) {
+			switch event.Type {
+			case watch.Modified:
+				if pod, ok := event.Object.(*v1.Pod); ok {
+					found := pod.ObjectMeta.Name == pod.Name &&
+						pod.Labels["test-pod-static"] == "true"
+					return found, nil
+				}
+			default:
+				framework.Logf("observed event type %v", event.Type)
+			}
+			return false, nil
+		})
+		framework.ExpectNoError(err, "failed to see %v event", watch.Modified)
+
+		ginkgo.By("getting the Pod and ensuring that it's patched")
+		pod, err := f.ClientSet.CoreV1().Pods(testNamespaceName).Get(context.TODO(), testPodName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "failed to fetch Pod %s in namespace %s", testPodName, testNamespaceName)
+		framework.ExpectEqual(pod.ObjectMeta.Labels["test-pod"], "patched", "failed to patch Pod - missing label")
+		framework.ExpectEqual(pod.Spec.Containers[0].Image, testPodImage2, "failed to patch Pod - wrong image")
+
+		ginkgo.By("getting the PodStatus")
+		podStatusUnstructured, err := dc.Resource(podResource).Namespace(testNamespaceName).Get(context.TODO(), testPodName, metav1.GetOptions{}, "status")
+		framework.ExpectNoError(err, "failed to fetch PodStatus of Pod %s in namespace %s", testPodName, testNamespaceName)
+		podStatusBytes, err := json.Marshal(podStatusUnstructured)
+		framework.ExpectNoError(err, "failed to marshal unstructured response")
+		var podStatus v1.Pod
+		err = json.Unmarshal(podStatusBytes, &podStatus)
+		framework.ExpectNoError(err, "failed to unmarshal JSON bytes to a Pod object type")
+
+		ginkgo.By("replacing the Pod's status Ready condition to False")
+		podStatusUpdated := podStatus
+		podStatusFieldPatchCount := 0
+		podStatusFieldPatchCountTotal := 2
+		for pos, cond := range podStatusUpdated.Status.Conditions {
+			if (cond.Type == v1.PodReady && cond.Status == v1.ConditionTrue) || (cond.Type == v1.ContainersReady && cond.Status == v1.ConditionTrue) {
+				podStatusUpdated.Status.Conditions[pos].Status = v1.ConditionFalse
+				podStatusFieldPatchCount++
+			}
+		}
+		framework.ExpectEqual(podStatusFieldPatchCount, podStatusFieldPatchCountTotal, "failed to patch all relevant Pod conditions")
+		podStatusUpdate, err := f.ClientSet.CoreV1().Pods(testNamespaceName).UpdateStatus(context.TODO(), &podStatusUpdated, metav1.UpdateOptions{})
+		framework.ExpectNoError(err, "failed to update PodStatus of Pod %s in namespace %s", testPodName, testNamespaceName)
+
+		ginkgo.By("check the Pod again to ensure its Ready conditions are False")
```

Comment:
Isn't there the possibility that the kubelet has already updated the pod's ready condition back to true here?

Reply:
I might be wrong, but from my understand the response from the API server (in this case `podStatusUpdate`) is basically the same as the request. If that 's true then it should be fine to do this, otherwise I'll remove the checks after the update (so the endpoint is hit)

Reply:
Ah, I misread the code, and thought we were doing a separate retrieval